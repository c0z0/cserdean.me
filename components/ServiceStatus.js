import React, { useState, useEffect } from "react";
import styled from "styled-components";

const apiRoot = "https://ai.cserdean.me";

const Ball = styled.div.attrs({ title: "AI Service Status" })`
  transition: all 0.2s linear;
  width: 10px;
  height: 10px;
  background: ${({ green }) => (green ? "#5ff441" : "#f4d142")};
  display: inline-block;
  border-radius: 100%;
`;

export default function ServiceStatus() {
  const [status, setStatus] = useState("init");

  useEffect(() => {
    (async () => {
      const res = await fetch(`${apiRoot}/ping`);
      if (res.ok) {
        setStatus("ready");
      }
    })();
  }, []);

  return <Ball green={status === "ready"} />;
}
