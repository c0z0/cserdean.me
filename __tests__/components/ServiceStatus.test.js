import React from "react";
import { render, cleanup } from "react-testing-library";
import "jest-styled-components";

import ServiceStatus from "../../components/ServiceStatus";

afterEach(cleanup);

test("Matches snapshot", () => {
  global.fetch = jest.fn(() => ({ ok: true }));

  const { container, rerender } = render(<ServiceStatus />);

  expect(container).toMatchSnapshot();

  rerender(<ServiceStatus />);
});
