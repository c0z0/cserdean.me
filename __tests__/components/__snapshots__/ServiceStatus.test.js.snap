// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
.c0 {
  -webkit-transition: all 0.2s linear;
  transition: all 0.2s linear;
  width: 10px;
  height: 10px;
  background: #f4d142;
  display: inline-block;
  border-radius: 100%;
}

<div>
  <div
    class="c0"
    title="AI Service Status"
  />
</div>
`;
